py/dml/ctree.py

```diff
@@ -3958,8 +3958,16 @@ def mkCast(site, expr, new_type):
                 expr.site,
```

Comment:
Oup, this path is unprotected! Needs checks here and a test to sniff it out.

Reply:
Actually, it's not unprotected, it just has a recursive call to `mkCast`. Still, it's good to have an explicit check to give better error messages (since the recursive call will have `uint64` as target type).